backend/controller/authController.js

```diff
@@ -52,12 +57,35 @@ const userExist = asyncHandler(async (req, res, next) => {
 const signUp = asyncHandler(async (req, res, next) => {
   const { email, password } = req.body;
 
+  if (!email || !password) {
+    return next(
+      new customError("All fields are required. email , password", 400)
+    );
+  }
+
+  const passwordRegex =
+    /^(?=.*[A-Z])(?=.*\d)(?=.*[!@#$%^&*()])[A-Za-z\d!@#$%^&*()]{6,60}$/;
+  if (!passwordRegex.test(password)) {
+    return next(
+      new customError(
+        "password must be 6 to 60 characters in length and contain at-least one capital letter, one symbol and one number",
+        400
+      )
+    );
+  }
+
   // check if the email is valid or not using email-validator npm package
   const isEmailValid = validator.validate(email);
 
   if (!isEmailValid)
     return next(new customError("Please enter a valid email 📩", 400));
 
+  const user = findOne({ email });
+  if (user)
+    return next(
+      new customError(`user with email: ${email} already exist`, 400)
```

Comment:
A better status code would be 409 in this case

Reply:
added